python/ucxx/_lib/libucxx.pyx

```diff
@@ -1302,6 +1303,7 @@ cdef class UCXListener():
         shared_ptr[Listener] _listener
         bint _enable_python_future
         dict _cb_data
+        object __weakref__
```

Comment:
Not sure I understand the purpose of this, could you explain its purpose?

Reply:
By default, `cdef` classes are not weakrefable (because they're just a struct, and so the `__weakref__` attribute doesn't exist). To enable it, we need this slot (https://cython.readthedocs.io/en/latest/src/userguide/extension_types.html#making-extension-types-weak-referenceable). This is kind of morally equivalent to defining a Python class with `__slots__`, at which point you must explicitly put `__weakref__` in slots to make the class weakrefable.

Reply:
Interesting, I wasn't aware of that. Thanks for the details.